File: 알고리즘/이것이코딩테스트다스터디/문자열/문자열폭발.py
# https://www.acmicpc.net/problem/9935
char = input()
bumpchar = input()
while bumpchar in char:
    char = char.replace(bumpchar, "")
    
if char == "":
    print("FRULA")
else:
    print(char)

length = len(bumpchar)
stack = []
for i in range(len(char)):
    stack.append(char[i])
    if bumpchar == ''.join(stack[-length:]):
        for _ in range(length):
            stack.pop()

newstr = ''.join(stack)
if newstr == "":
    print("FRULA")
else:
    print(newstr)

File: Algorithm/C++/백준/자료구조/문자열 폭발.cpp
// https://www.acmicpc.net/problem/9935

#include <iostream>
#include <string>
#include <stack>
using namespace std;
string words;
string explode;

int main()
{  
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    cin >> words >> explode;
    int n = words.size();
    int m = explode.size();
    string s;

    for(int i = 0; i < n; i ++) {
        s.push_back(words[i]);

        int cnt = s.size();
        while(s.size() >= m && explode == s.substr(cnt - m, m)) {
            for(int j = 0; j < m; j ++) s.pop_back();
        }
    } 

    if(s.size() == 0) cout << "FRULA";
    else cout << s;
}